Controllers/Backend/ConnectBaseController.php

```diff
@@ -626,27 +624,34 @@ public function loginAction(){
 
         $responseObject = json_decode($response->getBody());
 
-        if($responseObject->success) {
-            // Save the data
+        if(!$responseObject->success) {
+            $this->View()->assign([
+                'success' => false,
+                'message' => $responseObject->reason
+            ]);
+
+            return;
+        }
+
+        try {
             $this->getConfigComponent()->setConfig('apiKey', $responseObject->apiKey, null, 'general');
-            $this->removeConnectMenuEntry();
             $this->getSDK()->verifySdk();
             $this->getConfigComponent()->setConfig('apiKeyVerified', true);
             $this->getConfigComponent()->setConfig('shopwareId', $shopwareId);
+            $this->removeConnectMenuEntry();
             $marketplaceSettings = $this->getSDK()->getMarketplaceSettings();
             $this->getMarketplaceApplier()->apply(new MarketplaceSettings($marketplaceSettings));
+        } catch (\Exception $e) {
+            $this->getConfigComponent()->setConfig('apiKey', null, null, 'general');
```

Comment:
please move set api key after ```$this->getSDK()->verifySdk();``` then you don't need to set null if something went wrong.